### DISCUSSION TITLE
Are build-time-styles an opt-in feature?

### DISCUSSION THREAD

**Discussion by Wezea** [microsoft/fluentui#28844] [Q&A] [answered]
Hello folks,

we want to migrate our tec stack from CRA to vite. While going through the migration guide, we stumbled over the topic [build time styles](https://react.fluentui.dev/?path=/docs/concepts-developer-build-time-styles--page).

Our tec stack contains our own component library based on fluentui, which holds our basic theme context and a bunch of layout components. For the migration we also want to use it to provide shims and a kind of theme-context-hybrid out of v8 `ThemeProvider` and `FluentProvider`. 

The question would be, could we use vite as our bundler, even though griffel.js  does not support [build time compilation](https://github.com/microsoft/griffel/issues/33) yet and is this just an additional feature we could opt-in during our migration as soon as this is available?

**Comment by layershifter** [comment, marked as answer]
Yes, the feature is opt-in. You can use Fluent UI React v9 and `makeStyles` without any additional setup.

**Reply by Wezea** [reply]
Great! Thanks!
